config/samples/function-sample-with-path-based-route.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: core.openfunction.io/v1beta1
+kind: Function
+metadata:
+  name: function-sample-with-path-based-route
+spec:
+  version: "v1.0.0"
+  image: "openfunctiondev/v1beta1-http:latest"
+  port: 8080 # default to 8080
+  serving:
+    runtime: knative
+    template:
+      containers:
+        - name: function
+          imagePullPolicy: Always
+  route:
+    gatewayRef:
```

Comment:
the filename is path based route, but actually, the route is host based

Reply:
This is path-based route for `unction-sample-with-path-based-route.yaml`
![image](https://user-images.githubusercontent.com/18525465/180742752-1647d995-3015-4cd6-9275-e1b83b65e69c.png)

Reply:
If the route contains hostname, then the hostname is no longer generated according to the `hostTemplate`, only path will be generated according to the `pathTemplate`, so it is path-based, e.g:
`http://sample.ofn.io/default/function-sample`

Reply:
I'm thinking maybe the path-based method should be the default method to access function while the user needs to turn on a switch or add an explicit host option to enable host-based access

Reply:
I considered using path-based by default before, but the `domain`  in the gateway supports wildcards. using path-based by default means that we need to convert the wildcard domain name to fqdn and set default hostname, e.g: `dev.ofn.io`. after comparison, it is more flexible to use host-based by default.

Reply:
I can think of 3 ways to access a function:

- For users to access a function from outside the cluster, the user needs to set the gateway's service to load balancer, and he'll use the gateway IP to access the function. In this case, the path-based method is the default and the user can enable the host-based method if he wants
`curl http://<gateway ip>/<namespace>/<function name>`

- For users to access a function from inside the cluster's pod, the cluster FQDN is more suitable `<function name>.<namespace>.svc.cluster.local`

- For users to access a function from the cluster's node, he can use `curl http://<gateway ip>/<namespace>/<function name>` or the `curl http://samples.ofn.io/<namespace>/<function name>` while the later is recommened

Reply:
ok, if consider the use ip to access function, path-based is indeed more applicable than host-based(ip+headers).
will update this logic。

Reply:
This is not urgent, you can finishes the tasks to release v0.7.0 before working on this